### PR TITLE
add support for pagination hook

### PR DESCRIPTION
**Pull request by sarkistlt**
fix for #74

### CODE REVIEW

**Comment by idealley** [comment]
I will check the PR ASAP. Thank you for submitting.

**Comment by sarkistlt** [comment]
with this PR we may also need to update how clearSingle parses key from url

**Comment by Sicria** [comment]
Once this is merged #63 can be closed as well.

File: src/hooks/helpers/path.js [hunk 1]
```diff
@@ -15,12 +15,13 @@ function parseNestedPath(path, params) {
 
 function parsePath(hook, config = {removePathFromCacheKey: false, parseNestedRoutes: false}) {
   const q = hook.params.query || {};
+  const paginate = hook.params.paginate === false ? 'off' : 'on'; // if `.paginate` is underfined it means pagination hook is enabled, that's why we are using strict check with false
   const remove = config.removePathFromCacheKey;
   const parseNestedRoutes = config.parseNestedRoutes;
-  let path = remove && hook.id ? '' : `${hook.path}`;
+  let path = remove && hook.id ? `paginate:${paginate}:` : `paginate:${paginate}:${hook.path}`;
```
Comment (Sicria) [inline src/hooks/helpers/path.js]: This value could be assigned to a variable to reduce duplicate code.

File: src/hooks/helpers/path.js [hunk 2]
```diff
@@ -15,12 +15,13 @@ function parseNestedPath(path, params) {
...
 
   if (!remove && parseNestedRoutes) {
-    path = parseNestedPath(path, hook.params);
+    path = `paginate:${paginate}:${parseNestedPath(path, hook.params)}`;
   }
```
Comment (Sicria) [inline src/hooks/helpers/path.js:25]: You could move this to after the processing since ``parseNestedPath`` would ignore the value right?

**Pull request closed without merging**